upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -202,7 +202,10 @@ func (c *ApplyClusterCmd) Run() error {
 		if components.IsBaseURL(cluster.Spec.KubernetesVersion) {
 			baseURL = cluster.Spec.KubernetesVersion
 		} else {
-			baseURL = "https://storage.googleapis.com/kubernetes-release/release/v" + cluster.Spec.KubernetesVersion
+			baseURL, err = components.GetGoogleFileRepositoryURL(&c.Cluster.Spec, "/release/v"+cluster.Spec.KubernetesVersion)
```

Comment:
There's also the potential to run _everything_ though S3 - no need for a docker registry.  OTOH one step at a time, OTOH now users need a docker registry... I do wonder if it would be easier just to push everything to S3.